# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/res/cs_svg_filter.glsl
# Commit: 8d13c0ab66f0
# Full Hash: 8d13c0ab66f0ac137b1461d3451ecb9361ec2d5e
# Author: Jamie Nicol <jnicol@mozilla.com>
# Date: 2023-03-22 21:15:54
# Regressor Bug: 1823411
# File Overlap Count: 1
# Description:
#   Bug 1823411 - Give all webrender's shader varyings an explicit precision. r=gfx-reviewers,lsalzman
#   
#   Mali profiling tools have shown we are frequently fragment bound, in
#   particular due to varying interpolation. To help mitigate this, we
#   should use mediump where possible. Currently most of our varyings are
# ==============================================================================

diff -r 3919fbd972a6 -r 8d13c0ab66f0 gfx/wr/webrender/res/cs_svg_filter.glsl
--- a/gfx/wr/webrender/res/cs_svg_filter.glsl	Tue Mar 21 19:04:22 2023 +0000
+++ b/gfx/wr/webrender/res/cs_svg_filter.glsl	Tue Mar 21 19:05:19 2023 +0000
@@ -6,24 +6,24 @@
 
 #include shared,prim_shared
 
-varying vec2 vInput1Uv;
-varying vec2 vInput2Uv;
-flat varying vec4 vInput1UvRect;
-flat varying vec4 vInput2UvRect;
-flat varying ivec4 vData;
-flat varying vec4 vFilterData0;
-flat varying vec4 vFilterData1;
+varying highp vec2 vInput1Uv;
+varying highp vec2 vInput2Uv;
+flat varying highp vec4 vInput1UvRect;
+flat varying highp vec4 vInput2UvRect;
+flat varying mediump ivec4 vData;
+flat varying mediump vec4 vFilterData0;
+flat varying mediump vec4 vFilterData1;
 
 // x: Filter input count, y: Filter kind.
 // Packed in to a vector to work around bug 1630356.
-flat varying ivec2 vFilterInputCountFilterKindVec;
+flat varying mediump ivec2 vFilterInputCountFilterKindVec;
 #define vFilterInputCount vFilterInputCountFilterKindVec.x
 #define vFilterKind vFilterInputCountFilterKindVec.y
 // Packed in to a vector to work around bug 1630356.
-flat varying vec2 vFloat0;
+flat varying mediump vec2 vFloat0;
 
-flat varying mat4 vColorMat;
-flat varying ivec4 vFuncs;
+flat varying mediump mat4 vColorMat;
+flat varying mediump ivec4 vFuncs;
 
 #define FILTER_BLEND                0
 #define FILTER_FLOOD                1